### PR TITLE
KIN-343 integrate about us section with the admin panel

### PR DESCRIPTION
**Pull request by FernandoJoachin**
[KIN-343](https://loopcrack.atlassian.net/browse/KIN-343) Small [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=loopcrack-org&utm_campaign=badge_info)

### Problem

It is needed to integrate the admin panel generated values with the public view

### Solution

Integrate the about us record generated on the admin panel in the about us section of the public view

### Requirements checklist

- [x] The about us text generated on the admin panel must be displayed on the public view
- [x] The about us image stored through the admin panel must be displayed on its public view section.
- [x] The about us video stored through the admin panel must be displayed on its public view section.
- [x] The about us styles section layout should not break with the integration of the file values.

### Testing

- [x] Go to the next route /admin/nosotros and upload the corresponding files.
- [x] Go to the home page and check that the previously uploaded files are reflected on the site

[KIN-343]: https://loopcrack.atlassian.net/browse/KIN-343?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by CarlosMay7 — APPROVED** [review]
Nice job!

**Review by Cristianpan — APPROVED** [review]

**Review by CarlosJCD — CHANGES_REQUESTED** [review]
I think that it is important to limit the amount of text that the user can insert for the about  us text section on the admin panel. This is important because if it is not restricted then the user can place a lot of info and the layout of the about us section get broken. Please check it out and set a limit at your discretion

**Review by CarlosJCD — CHANGES_REQUESTED** [review]
There is an issue with the current way of the restriction of characters on the text field of the about us page. Because there is a limit of characters stablished on the html tag of the text area, the user gets restricted but there is no kind of feedback or warning telling the user that it was reached the limit of characters, even where it is a validation there is also no kind of tool or tip to the user indicating if the text has reached its limit or not. Maybe velzon has some kind of input that could help you with this.

CC: @CheloChidoone

**Review by CheloChidoone — CHANGES_REQUESTED** [review]
Good Job in general @FernandoJoachin but check @CarlosJCD comment about the issues with the text area character limit

**Review by LUGAMAFE — APPROVED** [review]
Hey @FernandoJoachin nice job can you set max dims for the image please. Im thinking about W1200px x H1000px

File: src/admin/js/pages/textareaCounter.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    characterCount.classList.add('d-block');
+    errorAboutUsText.classList.add('d-none');
+    errorAboutUsText.classList.remove('d-block');
+  }
+}
+
+function handleCharacterLimit() {
+  const textLength = textArea.value.length;
+
+  characterCount.textContent = `${textLength}/${450}`;
```
Comment (CarlosJCD) [inline src/admin/js/pages/textareaCounter.js:23]: I think it is only needed the `${}` syntax on string templates for operations or variables

```suggestion
  characterCount.textContent = `${textLength}/450`;
```

**Review by CarlosJCD — APPROVED** [review]
Nice work!. Just a nitpick

**Review by CheloChidoone — APPROVED** [review]

File: src/admin/js/pages/textareaCounter.js
Comment (LUGAMAFE) [inline src/admin/js/pages/textareaCounter.js:1]: hey @FernandoJoachin do we really need this file? if we already have in the textarea a maxlength="450"

File: src/admin/js/pages/textareaCounter.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  if (errorAboutUsText) {
+    let isError = textLength > 450;
+    updateVisibility(isError);
+  }
```
Comment (LUGAMAFE) [inline src/admin/js/pages/textareaCounter.js:28]: I dont think we need this code
```suggestion
```

File: src/admin/js/pages/textareaCounter.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// Get the textarea element and other required elements
+const textArea = document.getElementById('aboutUsText');
+const characterCount = document.getElementById('characterCount');
+const errorAboutUsText = document.querySelector('.invalid-feedback');
```
Comment (LUGAMAFE) [inline src/admin/js/pages/textareaCounter.js:4]: Where we have this element 
```suggestion
```

File: src/admin/js/pages/textareaCounter.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+function updateVisibility(isError) {
+  if (isError) {
+    characterCount.classList.remove('d-block');
+    characterCount.classList.add('d-none');
+    errorAboutUsText.classList.add('d-block');
+    errorAboutUsText.classList.remove('d-none');
```
Comment (LUGAMAFE) [inline src/admin/js/pages/textareaCounter.js:11]: remove 
```suggestion
```

File: src/admin/js/pages/textareaCounter.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  } else {
+    characterCount.classList.remove('d-none');
+    characterCount.classList.add('d-block');
+    errorAboutUsText.classList.add('d-none');
+    errorAboutUsText.classList.remove('d-block');
```
Comment (LUGAMAFE) [inline src/admin/js/pages/textareaCounter.js:16]: remove

**Review by LUGAMAFE — CHANGES_REQUESTED** [review]

**Review by Mauriciocr207 — APPROVED** [review]
Good job!

**Review by LUGAMAFE — CHANGES_REQUESTED** [review]
Hey @FernandoJoachin check these padding problems please
image

And can you help meGPT adding the capability to edit the "About Us" text.
image

**Comment by FernandoJoachin** [comment]
> Hey @FernandoJoachin check these padding problems please image
> 
> And can you help meGPT adding the capability to edit the "About Us" text. image

The padding problem was fixed in the KIN-128

File: app/Validation/AboutUsValidation.php [hunk 6]
```diff
@@ -5,11 +5,17 @@
 class AboutUsValidation extends BaseValidation
 {
     protected $validationRules = [
-        'aboutUsText' => 'required',
+        'aboutUsTitle' => 'required|max_length[20]',
+        'aboutUsText'  => 'required|max_length[330]',
     ];
     protected $validationMessages = [
+        'aboutUsTitle' => [
+            'required'   => 'El titulo es obligatoria.',
```
Comment (LUGAMAFE) [inline app/Validation/AboutUsValidation.php:13]: Nit
```suggestion
            'required'   => 'El titulo es obligatorio.',
```

**Review by Oswaldo-Chan — APPROVED** [review]

**Review by LUGAMAFE — APPROVED** [review]
All good @FernandoJoachin thank you! i just have a small nit

**Review by Oswaldo-Chan — APPROVED** [review]

**Pull request merged**